map/discovery_manager.cpp

```diff
@@ -0,0 +1,64 @@
+#include "map/discovery_manager.hpp"
+
+namespace
+{
+std::string GetQuery(discovery::ItemType const type)
+{
+  switch (type)
+  {
+  case discovery::ItemType::Hotels: return "hotel ";
+  case discovery::ItemType::Attractions: return "attraction ";
+  case discovery::ItemType::Cafe: return "cafe ";
+  case discovery::ItemType::LocalExperts:
+  case discovery::ItemType::Viator: ASSERT(false, ()); return "";
+  }
+}
+}  // namespace
+
+namespace discovery
+{
+Manager::Manager(Index const & index, search::CityFinder * const cityFinder, APIs const & apis)
+  : m_index(index)
+  , m_cityFinder(cityFinder)
+  , m_searchApi(apis.m_search)
+  , m_viatorApi(apis.m_viator)
+  , m_localsApi(apis.m_locals)
+{
+  CHECK(m_cityFinder, ());
+  CHECK(m_searchApi, ());
+  CHECK(m_viatorApi, ());
+  CHECK(m_localsApi, ());
+}
+
+// static
+search::SearchParams Manager::GetSearchParams(Manager::Params const & params, ItemType const type)
+{
+  search::SearchParams p;
+  p.m_query = GetQuery(type);
+  p.m_inputLocale = params.m_lang;
```

Comment:
input locale should be "en", right?

Reply:
I thought it should be what ```languages::GetCurrentNorm()``` returns, didn't it?

Reply:
On what language the query is written?

Reply:
On Android, the method languages::GetCurrentNorm() returns the default locale instead of the intput locale. Be aware of it please. If you need the input locale you should change the code a bit.